torch/csrc/distributed/c10d/symm_mem/nvshmem_extension.cu

```diff
@@ -107,6 +107,23 @@ void nvshmem_put(at::Tensor& tensor, const int64_t peer) {
   nvshmemx_putmem_on_stream(buffer_ptr, tensor.data_ptr(), buffer_size, peer, stream);
 }
 
+void nvshmem_wait_for_signal(at::Tensor& sigpad, int64_t signal, int64_t peer) {
+  c10::cuda::CUDAGuard guard(sigpad.device());
+  auto stream = at::cuda::getCurrentCUDAStream();
+  nvshmemx_signal_wait_until_on_stream(static_cast<uint64_t*>(sigpad.data_ptr()), NVSHMEM_CMP_EQ, signal, stream);
+}
+
+void nvshmem_put_with_signal(at::Tensor& tensor, at::Tensor& sigpad, int64_t signal, int64_t peer) {
+  auto hdl = c10d::symmetric_memory::rendezvous(tensor, "0");
```

Comment:
Maybe we should remove this rendezvous call, because not all ranks are calling this put API -- this one here would hang.
Also we don't need anything from the handle actually?